pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/SpecialValueTransformer.java

```diff
@@ -18,112 +18,119 @@
  */
 package org.apache.pinot.segment.local.recordtransformer;
 
-import com.google.common.annotations.VisibleForTesting;
+import java.math.BigDecimal;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
 import org.apache.pinot.spi.data.FieldSpec;
-import org.apache.pinot.spi.data.FieldSpec.DataType;
 import org.apache.pinot.spi.data.Schema;
 import org.apache.pinot.spi.data.readers.GenericRow;
 import org.apache.pinot.spi.recordtransformer.RecordTransformer;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 
 /**
  * The {@code SpecialValueTransformer} class will transform special values according to the following rules:
  * <ul>
- *   <li>Negative zero (-0.0) should be converted to 0.0</li>
- *   <li>NaN should be converted to default null</li>
+ *   <li>
+ *     For FLOAT and DOUBLE:
+ *     <ul>
+ *       <li>Negative zero (-0.0) should be converted to 0.0</li>
+ *       <li>NaN should be converted to default null</li>
+ *     </ul>
+ *   </li>
+ *   <li>
+ *     For BIG_DECIMAL:
+ *     <ul>
+ *       <li>Remove trailing zeros</li>
```

Comment:
nit: if we're going to leave a comment, we might as well say why so folks don't have to come back to the PR